Fuzzer can't generate values (even without any modifications) when there are some modifications it fails to use

**Description (abstract example)**

When using `Seed.Recursive` with modification `X` that has a parameter of type `T`, while there are no seeds for type `T`, fuzzer doesn't generate any values for that `Seed.Recursive` (except for the ones produced by its `Routine.Empty`, which is typically just `null`)

**Real world (concrete) example**

No integration tests using `topicRepository.save(Topic)` modification are generated for methods of [`TopicService` class in `Medical-Web-App` project](https://github.com/MathAndMedLab/Medical-Web-App/blob/63ed67c73466eacea25e1a6335091e0e261fd535/src/main/java/com/app/medicalwebapp/services/TopicService.java) (or any other class that depends on `TopicRepository`). That happens because:
1. There's `Topic.setCreationTime(LocalDateTime)` modification.
2. Java fuzzer can't generate any values for `LocalDateTime` (_a separate issue_).
3. Fuzzing platform fails to generate non-`null` values for `Topic` type due to point 2 and 3 (**concern of this issue**).
4. Modification `topicRepository.save(Topic)` is only used with `null` values of `topic`, which causes modification itself to fail before we even get to method under test.

**To Reproduce**

Run the following unit test.
```kotlin
fun `fuzzer can generate non empty values even when it can't modify it due to lack of seeds`() {
   class Type(val hasSeeds: Boolean)
   class Value(val isEmpty: Boolean)

   runBlocking {
       withTimeout(1000) {
           runFuzzing(
               { _, type ->
                   if (type.hasSeeds) sequenceOf(Seed.Recursive(
                       construct = Routine.Create(emptyList()) { Value(isEmpty = false) },
                       modify = sequenceOf(Routine.Call(listOf(Type(hasSeeds = false))) { _, _ ->
                           fail("Value is generated for type with no seeds")
                       }),
                       empty = Routine.Empty { Value(isEmpty = true) }
                   ))
                   else emptySequence()
               },
               Description(listOf(Type(hasSeeds = true)))
           ) { _, (value) ->
               if (value.isEmpty) BaseFeedback(result = Unit, Control.CONTINUE)
               else BaseFeedback(result = Unit, Control.STOP)
           }
       }
   }
}
```

**Expected behavior**

Test passes.

**Actual behavior**

Test fails.

**Visual proofs (screenshots, logs, images)**

```
Timed out waiting for 1000 ms
kotlinx.coroutines.TimeoutCancellationException: Timed out waiting for 1000 ms
    (Coroutine boundary)
    at org.utbot.fuzzing.FuzzerSmokeTest$fuzzer can generate non empty values even when it can't modify it due to lack of seeds$1.invokeSuspend(FuzzerSmokeTest.kt:445)
Caused by: kotlinx.coroutines.TimeoutCancellationException: Timed out waiting for 1000 ms
    at app//kotlinx.coroutines.TimeoutKt.TimeoutCancellationException(Timeout.kt:184)
    at app//kotlinx.coroutines.TimeoutCoroutine.run(Timeout.kt:154)
    at app//kotlinx.coroutines.EventLoopImplBase$DelayedRunnableTask.run(EventLoop.common.kt:508)
    at app//kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
    at app//kotlinx.coroutines.DefaultExecutor.run(DefaultExecutor.kt:108)
    at java.base@19.0.2/java.lang.Thread.run(Thread.java:1589)
```

--- a/utbot-fuzzing/src/main/kotlin/org/utbot/fuzzing/Api.kt
+++ b/utbot-fuzzing/src/main/kotlin/org/utbot/fuzzing/Api.kt
@@ -422,7 +422,7 @@ private fun <TYPE, RESULT, DESCRIPTION : Description<TYPE>, FEEDBACK : Feedback<
     if (seeds.isEmpty()) {
         throw NoSeedValueException(type)
     }
-    val candidates = seeds.map {
+    return seeds.random(random).let {
         when (it) {
             is Seed.Simple<TYPE, RESULT> -> Result.Simple(it.value, it.mutation)
             is Seed.Known<TYPE, RESULT, *> -> it.asResult()
@@ -430,7 +430,6 @@ private fun <TYPE, RESULT, DESCRIPTION : Description<TYPE>, FEEDBACK : Feedback<
             is Seed.Collection<TYPE, RESULT> -> reduce(it, fuzzing, description, random, configuration, state)
         }
     }
-    return candidates.random(random)
 }
 
 /**

--- a/utbot-fuzzing/src/test/kotlin/org/utbot/fuzzing/FuzzerSmokeTest.kt
+++ b/utbot-fuzzing/src/test/kotlin/org/utbot/fuzzing/FuzzerSmokeTest.kt
@@ -5,9 +5,11 @@ import kotlinx.coroutines.flow.flow
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Assertions.fail
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Timeout
 import org.junit.jupiter.api.assertThrows
 import org.utbot.fuzzing.seeds.BitVectorValue
 import org.utbot.fuzzing.seeds.Signed
+import java.util.concurrent.TimeUnit
 import kotlin.reflect.KClass
 
 class FuzzerSmokeTest {
@@ -280,6 +282,30 @@ class FuzzerSmokeTest {
                 }
             }
             Assertions.assertTrue(seenEmpty) { "Unmodified empty string wasn't generated" }
+        }
+    }
+
+    @Test
+    @Timeout(10, unit = TimeUnit.SECONDS) // withTimeout(1000) works inconsistently
+    fun `fuzzer works when there are many recursive seeds`() {
+        class Node(val parent: Node?)
+
+        runBlocking {
+            var seenAnything = false
+            withTimeout(1000) {
+                runFuzzing(
+                    { _, _ -> List(100) {Seed.Recursive<Unit, Node?>(
+                        construct = Routine.Create(listOf(Unit)) { (parent) -> Node(parent) },
+                        modify = emptySequence(),
+                        empty = Routine.Empty { null }
+                    )}.asSequence() },
+                    Description(listOf(Unit))
+                ) { _, _ ->
+                    seenAnything = true
+                    BaseFeedback(Unit, Control.STOP)
+                }
+            }
+            Assertions.assertTrue(seenAnything) { "Fuzzer hasn't generated any values" }
         }
     }
 }

